crates/contracts/solidity/Trader.sol

```diff
@@ -1,11 +1,80 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.16;
 
-import { IERC20, INativeERC20 } from "./interfaces/IERC20.sol";
-import { Interaction } from "./interfaces/ISettlement.sol";
+import { IERC20, IMintableERC20, INativeERC20 } from "./interfaces/IERC20.sol";
+import { Interaction, Trade, SETTLEMENT } from "./interfaces/ISettlement.sol";
+import { Caller } from "./libraries/Caller.sol";
+import { SafeERC20 } from "./libraries/SafeERC20.sol";
 
 /// @title A contract for impersonating a trader.
 contract Trader {
+    using Caller for *;
+    using SafeERC20 for *;
+
+    /// @dev Simulates a executing a trade with the CoW protocol settlement
+    /// contract. This sort of simulation provides stronger guarantees that the
+    /// proposed trade is valid and would work in an actual settlement.
+    ///
+    /// @param tokens - tokens included in the settlement. Balances will be
+    /// tracked for each token included in this array. `tokens[0]` is the trade
+    /// sell token and `tokens[tokens.length - 1]` is the buy token.
+    /// @param clearingPrices - the clearing prices for the settlement. This
+    /// also doubles as the traded amounts, with `clearingPrices[0]` being the
+    /// buy amount and the `clearingPrices[tokens.length - 1]` the sell amount.
+    /// @param interactions - the interactions for settling the trade.
+    /// @param mint - mint some sell token if this is a non-zero value. This
+    /// requires that the sell token is mintable (which can be guaranteed by
+    /// replacing its code with the `PhonyERC20` contract).
+    function settle(
```

Comment:
Could you also add documentation for the return values? It makes sense after looking at implementation but when I only read the signature I'm not sure what the balances mean.

Reply:
OK - will do. There should be a NatSpec way to do this.

Reply:
I don't know what NatSpec is. I'd be fine just putting it in the documentation comment that is already there even if documentation generator wouldn't pick it up.

Reply:
NatSpec is just the standard Solidity documentation format.

Agree that if it wasn't supported I could just add it to the comments as "normal" documentation.